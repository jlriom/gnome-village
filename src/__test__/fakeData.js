export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
