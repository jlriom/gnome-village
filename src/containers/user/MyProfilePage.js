import React from "react";
import { connect } from "react-redux";
import {
	saveProfile,
	loadHabitantsProfessions,
	loadHabitantsHairColors,
	loadProfile
} from "../../store/actions/userActions";
import HabitantDetailsEdit from "../../components/HabitantDetailsEdit";

const MyProfilePage = ({
	id,
	profile,
	professions,
	hairColors,
	error,
	show,
	handleClose,
	loadProfile,
	loadHabitantsProfessions,
	loadHabitantsHairColors,
	saveProfile
}) => {
	return (
		<HabitantDetailsEdit
			mode={{ mode: "edit", title: "My profile", submitCaption: "Save" }}
			id={id}
			profile={profile}
			professions={professions}
			hairColors={hairColors}
			show={show}
			handleClose={handleClose}
			loadProfile={loadProfile}
			loadHabitantsProfessions={loadHabitantsProfessions}
			loadHabitantsHairColors={loadHabitantsHairColors}
			saveProfile={saveProfile}
		/>
	);
};

const mapStateToProps = state => ({
	id: state.auth.id,
	profile: state.user.profile,
	professions: state.user.professions,
	hairColors: state.user.hairColors,
	error: state.user.error
});

export default connect(
	mapStateToProps,
	{
		loadProfile,
		loadHabitantsProfessions,
		loadHabitantsHairColors,
		saveProfile
	}
)(MyProfilePage);
