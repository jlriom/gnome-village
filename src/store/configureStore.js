import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
