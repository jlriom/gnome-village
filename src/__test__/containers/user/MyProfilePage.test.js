import React from "react";
import MyProfilePage from "../../../containers/user/MyProfilePage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "../../../store/configureStore";
import { habitant } from "../../fakeData";

const store = configureStore();

const props = {
    id: 0,
    profile: habitant,
    professions: [],
    hairColors: [],
    show: false,
    handleClose: jest.fn(),
    loadProfile: jest.fn(),
    loadHabitantsProfessions: jest.fn(),
    loadHabitantsHairColors: jest.fn(),
    saveProfile: jest.fn(),
    showError: jest.fn()
}

describe("<MyProfilePage />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MyProfilePage {...props} />
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});
