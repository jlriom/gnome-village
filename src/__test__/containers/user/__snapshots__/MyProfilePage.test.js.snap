// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyProfilePage /> renders ok 1`] = `null`;
